man/fabric.7.md

```diff
@@ -300,23 +300,13 @@ portability across providers.
   on write restrictions.
 
 ## CUDA deadlock
-In some cases, calls to `cudaMemcpy` within libfabric may result in a
-deadlock. This typically occurs when a CUDA kernel blocks until a
-`cudaMemcpy` on the host completes.  To avoid this deadlock,
-`cudaMemcpy` may be disabled by setting
-`FI_HMEM_CUDA_ENABLE_XFER=0`. If this environment variable is set and
-there is a call to `cudaMemcpy` with libfabric, a warning will be
-emitted and no copy will occur. Note that not all providers support
-this option.
-
-Another mechanism which can be used to avoid deadlock is Nvidia's
-gdrcopy. Using gdrcopy requires an external library and kernel module
-available at https://github.com/NVIDIA/gdrcopy. Libfabric must be
-configured with gdrcopy support using the `--with-gdrcopy` option, and
-be run with `FI_HMEM_CUDA_USE_GDRCOPY=1`. This may be used in
-conjunction with the above option to provide a method for copying
-to/from CUDA device memory when `cudaMemcpy` cannot be used. Again,
-this may not be supported by all providers.
+In some cases, calls to `cudaMemcpy` within libfabric may result in a deadlock.
+This typically occurs when a CUDA kernel blocks until a `cudaMemcpy` on the host
+completes.  One mechanism which can be used to avoid deadlock is Nvidia's
+GDRCopy. Using GDRCopy requires an external library and kernel module available
+at https://github.com/NVIDIA/gdrcopy. Libfabric must be configured with GDRCopy
+support using the `--with-gdrcopy` option, and be run with
+`FI_HMEM_CUDA_USE_GDRCOPY=1`. This may not be supported by all providers.
```

Comment:
I think we should mention that if application have such behavior that may cause deadlock, it may use the FI_OPT_CUDA_API_PERMITTED to inform libfabric of such restriction.

Reply:
Does configure enable gdrcopy, if present, without requiring a separate configure option?

Reply:
> Does configure enable gdrcopy, if present, without requiring a separate configure option?


Yes.

Reply:
Since these lines are changing, I would capture something to that effect.  E.g. ```...using the --with-gdrcopy option (by default, configure will automatically attempt to detect if gdrcopy is present and usable) ...```

Reply:
Sorry I merged after your approval. 

@darrylabbate  Can you update the doc as @shefty  suggested?